File: artikAgentFactory/backend/app/schemas/notification_settings.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class NotificationSettingsOut(BaseModel):
    workspace_id: int
    slack_enabled: bool
    notify_on_run_completed: bool
    notify_on_new_results: bool
    notify_on_changed_results: bool
    notify_on_high_priority: bool
    notify_on_deadline_approaching: bool
    notify_on_run_error: bool
    min_severity: str
    slack_configured: bool
    updated_at: datetime


class NotificationSettingsUpdate(BaseModel):
    slack_enabled: bool | None = None
    notify_on_run_completed: bool | None = None
    notify_on_new_results: bool | None = None
    notify_on_changed_results: bool | None = None
    notify_on_high_priority: bool | None = None
    notify_on_deadline_approaching: bool | None = None
    notify_on_run_error: bool | None = None
    min_severity: str | None = None

File: artikAgentFactory/frontend/src/api/notificationSettings.ts
import { api } from "./client";

export type NotificationSettings = {
  workspace_id: number;
  slack_enabled: boolean;
  notify_on_run_completed: boolean;
  notify_on_new_results: boolean;
  notify_on_changed_results: boolean;
  notify_on_high_priority: boolean;
  notify_on_deadline_approaching: boolean;
  notify_on_run_error: boolean;
  min_severity: string;
  slack_configured: boolean;
  updated_at: string;
};

export const notificationSettingsApi = {
  get: () => api.get<NotificationSettings>("/api/notification-settings"),
  update: (body: Partial<NotificationSettings>) => api.put<NotificationSettings>("/api/notification-settings", body),
};
